Sources/WebService/AuthenticatedWebService/AuthenticatedWebService.swift

```diff
@@ -56,84 +56,83 @@ open class AuthenticatedWebService: WebService {
     self.configuration = configuration
     super.init(urlSession: urlSession)
   }
-  
+
   override public func execute<T>(urlRequest: URLRequest) -> AnyPublisher<T, Error> where T : Decodable {
     var urlRequest = urlRequest
-    var currentAccessToken: String?
 
-    authenticationQueue.sync {
-      currentAccessToken = self.tokenProvider.accessToken.value
+    func appendTokenAndExecute(accessToken: AccessToken) -> AnyPublisher<T, Error> {
+      urlRequest.setValue(self.configuration.authorizationHeaderScheme.rawValue + accessToken, forHTTPHeaderField: "Authorization")
+      return super.execute(urlRequest: urlRequest)
+        .subscribe(on: DispatchQueue.global())
+        .eraseToAnyPublisher()
+    }
+
+    return Deferred {
+      self.tokenProvider.accessToken
+        .compactMap { $0 }
+        .setFailureType(to: Error.self)
+        .flatMap { accessToken -> AnyPublisher<T, Error> in
+          return appendTokenAndExecute(accessToken: accessToken)
+      }
     }
-    
-    guard let accessToken = currentAccessToken else {
-      return Fail<T, Error>(error: NetworkError.unauthorized).eraseToAnyPublisher()
+    .timeout(10, scheduler: DispatchQueue.main, customError: { NetworkError.timeout })
```

Comment:
Again, I don't think this timeout and queue logic is responsibility of this framework

Reply:
if there is no timeOut, the request will hang forever. so it's an extra thing to think about for the consumer. Token refreshing is a core feature of this framework, which comes with the retry mechanism and so on. If we keep all in but the timeout outside, then the consumer has to think about this extra responsibility. Does it make a difference if it's done inside or outside?